src/mpid/ch4/generic/am/mpidig_am_send.h

```diff
@@ -70,18 +70,65 @@ MPL_STATIC_INLINE_PREFIX int MPIDIG_isend_impl(const void *buf, MPI_Aint count,
                                                MPIDI_av_entry_t * addr, uint8_t flags,
                                                MPIR_Request ** request, MPIR_Errflag_t errflag)
 {
-    MPI_Aint data_sz;
-    MPIR_Datatype *dt_ptr;
-    MPIDI_Datatype_get_size_dt_ptr(count, datatype, data_sz, dt_ptr);
-
-    int is_local = MPIDI_av_is_local(addr);
-    if (data_sz > mpidig_eager_limit(is_local)) {
-        return MPIDIG_do_rndv_send(buf, count, datatype, data_sz, rank, tag, comm, context_offset,
-                                   addr, flags, request, errflag);
+    int mpi_errno = MPI_SUCCESS;
+    MPIDIG_hdr_t send_hdr;
+    MPIR_Request *sreq = NULL;
+    size_t data_sz = 0;
+
+    MPIR_FUNC_VERBOSE_STATE_DECL(MPID_STATE_MPIDIG_ISEND_IMPL);
+    MPIR_FUNC_VERBOSE_ENTER(MPID_STATE_MPIDIG_ISEND_IMPL);
+
+    sreq = *request;
+    if (sreq == NULL) {
+        sreq = MPIDIG_request_create(MPIR_REQUEST_KIND__SEND, 2);
+        MPIR_ERR_CHKANDSTMT((sreq) == NULL, mpi_errno, MPIX_ERR_NOREQ, goto fn_fail, "**nomemreq");
+        *request = sreq;
     } else {
-        return MPIDIG_do_eager_send(buf, count, datatype, rank, tag, comm, context_offset,
-                                    addr, flags, request, errflag);
```

Comment:
`MPIDIG_do_rndv_send` and `MPIDIG_do_eager_send` should be removed, right?

Reply:
It is removed, the new code was added to `MPIDIG_isend_impl`. But the diff is matching it with `MPIDIG_do_eager_send` as they are the most similar.